ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/BlockProcessorUtil.java

```diff
@@ -81,632 +100,475 @@
     }
+  }
 
-    // If we're seeing this hash for the first time, make a new counter
-    state
-        .getEth1_data_votes()
-        .add(new Eth1DataVote(block.getBody().getEth1_data(), UnsignedLong.ONE));
+  // @v0.7.1
+  public static void process_operations(BeaconState state, BeaconBlockBody body)
+          throws BlockProcessingException {
+    try {
+
+      checkArgument(body.getDeposits().size() == Math.min(MAX_DEPOSITS, toIntExact(state.getLatest_eth1_data().getDeposit_count().minus(state.getDeposit_index()).longValue())),
+            "Verify that outstanding deposits are processed up to the maximum number of deposits");
+    Set transfersSet = new HashSet(body.getTransfers());
+    checkArgument(body.getTransfers().size() == transfersSet.size(), "Verify that there are no duplicate transfers");
+
+    process_proposer_slashings(state, body.getProposer_slashings());
+    process_attester_slashings(state, body.getAttester_slashings());
+    process_attestations(state, body.getAttestations());
+    process_deposits(state, body.getDeposits());
+    process_voluntary_exits(state, body.getVoluntary_exits());
+    process_transfers(state, body.getTransfers());
+
+    } catch (BlockProcessingException e) {
+      LOG.log(Level.WARN, "BlockProcessingException thrown in process_operations()");
+      throw e;
+    }
   }
 
-  /**
-   * Process ``ProposerSlashing`` transaction. Note that this function mutates ``state``.
-   *
-   * @param state
-   * @param block
-   * @throws BlockProcessingException
-   */
-  public static void process_proposer_slashings(BeaconState state, BeaconBlock block)
+  // @v0.7.1
+  public static void process_proposer_slashings(BeaconState state, List<ProposerSlashing> proposerSlashings)
       throws BlockProcessingException {
     try {
       // Verify that len(block.body.proposer_slashings) <= MAX_PROPOSER_SLASHINGS
       checkArgument(
-          block.getBody().getProposer_slashings().size() <= MAX_PROPOSER_SLASHINGS,
+          proposerSlashings.size() <= MAX_PROPOSER_SLASHINGS,
           "Proposer slashings more than limit in process_proposer_slashings()");
 
       // For each proposer_slashing in block.body.proposer_slashings:
-      for (ProposerSlashing proposer_slashing : block.getBody().getProposer_slashings()) {
-        // - Let proposer = state.validator_registry[proposer_slashing.proposer_index]
+      for (ProposerSlashing proposer_slashing : proposerSlashings) {
         Validator proposer =
             state
                 .getValidator_registry()
                 .get(toIntExact(proposer_slashing.getProposer_index().longValue()));
 
-        // Verify that the epoch is the same
         checkArgument(
             slot_to_epoch(proposer_slashing.getHeader_1().getSlot())
                 .equals(slot_to_epoch(proposer_slashing.getHeader_2().getSlot())),
-            "Epoch is not the same in process_proposer_slashings");
+            "Verify that the epoch is the same");
```

Comment:
I would prefer if you still keep the name of the method in the checkArgument errorMessage